src/org/loklak/api/cms/DumpDownloadServlet.java

```diff
@@ -1,160 +1,163 @@
-        if (dump == null || !dump.exists()) {
-            response.sendError(404, request.getContextPath() + " not available");
-        }
-
-        response.setDateHeader("Last-Modified", dump.lastModified());
-        response.setDateHeader("Expires", Math.max(now + ((now - dump.lastModified()) / 2), now + 600000));
-        response.setContentLengthLong(dump.length());
-        response.setContentType("application/octet-stream");
-        response.setCharacterEncoding("UTF-8");
-        response.setStatus(HttpServletResponse.SC_OK);
-        try {
-            InputStream fis = new BufferedInputStream(new FileInputStream(dump));
-            byte[] buffer = new byte[65536];
-            int c;
-            while ((c = fis.read(buffer)) > 0) {
-                response.getOutputStream().write(buffer, 0, c);
-            }
-            fis.close();
-        } catch (Throwable e) {
-            DAO.severe(e);
-        }
-        post.finalize();
-    }
-}
+/**
+ *  DumpDownloadServlet
+ *  Copyright 22.02.2015 by Michael Peter Christen, @0rb1t3r
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.loklak.api.cms;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStreamWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.Date;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.loklak.data.DAO;
+import org.loklak.http.RemoteAccess;
+import org.loklak.server.FileHandler;
+import org.loklak.server.Query;
+
+public class DumpDownloadServlet extends HttpServlet {
+
+    private static final long serialVersionUID = 2839106194602799989L;
+
+    @Override
+    protected void doPost(HttpServletRequest request,
+        HttpServletResponse response) throws ServletException, IOException {
+        doGet(request, response);
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest request,
+        HttpServletResponse response) throws ServletException, IOException {
+        if (!DAO.writeDump) {
```

Comment:
https://github.com/loklak/loklak_server/blob/9d06850a3e486ae02adc3b210f45afba34d01ee7/src/org/loklak/data/DAO.java#L776
this method also writes message dump. 
If flag is added here, The messages will not be dumped while messages are collected.

this servlet outputs all messages collected by above in html to client. the above method(in link) dumps them.